Jenkinsfile

```diff
@@ -5,46 +5,126 @@ pipeline {
     }
     options {
         buildDiscarder(logRotator(daysToKeepStr: '30'))
+        parallelsAlwaysFailFast()
     }
     triggers { cron(env.BRANCH_NAME ==~ /^\d+\.\d+\.x$/ ? 'H H(0-6) * * *' : '') }
     stages {
-        stage('BuildAndTest') {
-            matrix {
-                axes {
-                    axis {
-                        name 'FRAMEWORK'
-                        values 'akeneo', 'drupal8', 'magento1', 'magento2', 'php', 'spryker', 'symfony', 'wordpress'
+        stage('Build and Test') {
+            parallel {
+                stage('1. PHP, Drupal 8, Akeneo') {
+                    agent { label "my127ws" }
+                    stages {
+                        stage('Prepare') {
+                            steps { sh './build' }
```

Comment:
build can't be in the parallel stage, as it needs to be available before the parallel stage, otherwise the tests can easily fail through race conditions

Reply:
There's no agent selected until this stage, so nothing checked out. This is the first step on a worker post-checkout.

Reply:
so.. do you mean it pulls this stage out into a non-parallel stage automatically?

Reply:
oh, I see the only parallel bit is the stage('#. ....'), these are all sequential